### PR TITLE
[ROCm][CI] Fix fused RMS norm FP8 quant test on MI250 (gfx90a)

### PR DESCRIPTION
**Pull request by AndreasKaratzas**
`test_fused_rms_norm_quant` fails consistently on MI250 (gfx90a) for `dtype=half`, `add_residual=False`. The test compares the fused `rms_norm_static_fp8_quant` kernel against an unfused `rms_norm` + `static_scaled_fp8_quant` pipeline and expects
bit-identical FP8 output.

On gfx90a the fused kernel produces different FP8 values for roughly 0.3% of elements (9-18 out of millions at `num_tokens=4096`).

## Root cause

`c10::Half::operator*` does not use native half arithmetic. It promotes both operands to float, multiplies in float, then constructs a new `Half` from the result:

```cpp
// PyTorch c10/util/Half.h
inline C10_HOST_DEVICE Half operator*(const Half& a, const Half& b) {
  return static_cast<float>(a) * static_cast<float>(b);
}
```

In the fused kernel the expression `((scalar_t)(x * s_variance)) * src2.val[j]` feeds directly into a `float` variable. The half value returned by `operator*` is never stored to memory -- it exists only as a temporary. Under hipcc's default `-ffast-math`, the compiler on gfx90a is free to elide the float-to-half-to-float round-trip since the intermediate half is unobservable.

The unfused `rms_norm` kernel writes its result to a global-memory `scalar_t` (half) tensor, forcing the rounding. When the value sits exactly on an FP8 quantisation boundary the 1-ULP difference from the elided rounding shifts it into the adjacent FP8 bin.

This does not reproduce on MI300/MI325 (gfx942/gfx950) where the compiler makes different optimisation decisions for half arithmetic.

## Fix

Three scoped changes, all gfx90a-only:

**`csrc/layernorm_quant_kernels.cu`** -- For `c10::Half` on gfx90a, compute the normalised value in float explicitly (`x * s_variance * float(weight)`) instead of relying on `Half::operator*`. This makes the kernel's precision deterministic regardless of compiler optimisation, avoids two unnecessary float-to-half conversions, and is marginally faster and more accurate. Other dtypes (`bf16`, `float`) and other architectures are unchanged.

**`tests/kernels/core/test_layernorm.py`** -- On gfx90a + half the test builds a float32 reference (PyTorch RMS norm in float) that matches the kernel's float computation path. A small residual mismatch remains (~0.003% of elements, max 1 FP8 step)  because PyTorch's CUDA reduction and the kernel's CUB `BlockReduce` use different tree structures. This is validated with a structured precision check (same pattern as the AITER FP8 precision checks) instead of strict `assert_close`. All other dtype/arch combinations keep the original strict comparison. Also switches FP8 output buffers from `torch.empty` to `torch.zeros` for ROCm reliability. 

**`vllm/platforms/rocm.py`** -- Adds `on_gfx90a()` helper alongside the existing `on_gfx942()` / `on_gfx950()` helpers.

cc @micah-wil @kenroche

### CODE REVIEW

**Comment by AndreasKaratzas** [comment]
This is too deprecated.
